dagger/utilities/dbt_config_parser.py

```diff
@@ -0,0 +1,131 @@
+import json
+from os import path
+from os.path import join
+from pprint import pprint
+from typing import Tuple, List, Dict
+
+import yaml
+
+ATHENA_TASK_BASE = {"type": "athena"}
+S3_TASK_BASE = {"type": "s3"}
+
+
+
+class DBTConfigParser:
+    """
+    Module that parses the manifest.json file generated by dbt and generates the dagger inputs and outputs for the respective dbt model
+    """
+
+    def __init__(self, default_config_parameters: dict):
+        self._default_data_bucket = default_config_parameters["data_bucket"]
+        self._dbt_project_dir = default_config_parameters.get("project_dir", None)
+        dbt_manifest_path = path.join(self._dbt_project_dir, "target", "manifest.json")
+        self._dbt_profile_dir = default_config_parameters.get("profile_dir", None)
+        dbt_profile_path = path.join(self._dbt_profile_dir, "profiles.yml")
+
+        with open(dbt_manifest_path, "r") as f:
+            data = f.read()
+        self._manifest_data = json.loads(data)
+        profile_yaml = yaml.safe_load(open(dbt_profile_path, "r"))
+        prod_dbt_profile = profile_yaml[self._dbt_project_dir.split("/")[-1]][
+            "outputs"
+        ]["data"]
+        self._default_data_dir = prod_dbt_profile.get(
+            "s3_data_dir"
+        ) or prod_dbt_profile.get("s3_staging_dir")
+
```

Comment:
what is the difference btw `s3_data_dir` and `s3_staging_dir`?  It is fine that the `s3_data_dir` is empty?

Reply:
![image](https://github.com/chocoapp/dataeng-dagger/assets/16002904/87916c10-4adb-49cb-bf50-70d0cbd31661)

`s3_data_dir` is an optional parameter and if used, stores data in this directory. Since it is an optional parameter, it can be empty.